pancancer_utilities/scripts/classify_cancer_type.py

```diff
@@ -0,0 +1,244 @@
+p.add_argument('--holdout_cancer_type', type=str, required=True,
+               help='Provide a cancer type to hold out; train on all others.')
+p.add_argument('--num_folds', type=int, default=4,
+               help='Number of folds of cross-validation to run')
+p.add_argument('--use_pancancer', action='store_true',
+               help='Whether or not to use pan-cancer data in model training')
+p.add_argument('--results_dir', default=cfg.results_dir,
+               help='Where to write results to')
+p.add_argument('--seed', type=int, default=cfg.default_seed)
+p.add_argument('--shuffle_labels', action='store_true',
+               help='Include flag to shuffle labels as a negative control')
+p.add_argument('--verbose', action='store_true')
+args = p.parse_args()
+
+np.random.seed(args.seed)
+if args.verbose:
+    logging.basicConfig(level=logging.DEBUG, format='%(message)s')
+
+# load and unpack pancancer data
+genes_df, pancan_data = du.load_pancancer_data([args.gene], verbose=args.verbose)
+gene_idx = 0
+gene_name = genes_df.iloc[0, :].gene
+classification = genes_df.iloc[0, :].classification
+
+(sample_freeze_df,
+ mutation_df,
+ copy_loss_df,
+ copy_gain_df,
+ mut_burden_df) = pancan_data
+
+# load expression data
+rnaseq_df = du.load_expression_data(verbose=args.verbose)
+sample_info_df = du.load_sample_info(verbose=args.verbose)
+assert args.holdout_cancer_type in np.unique(sample_info_df.cancer_type), \
+        'Holdout cancer type must be a valid TCGA cancer type identifier'
+
+# track total metrics for each gene in one file
+metric_cols = [
+    "auroc",
+    "aupr",
+    "gene",
+    "holdout_cancer_type",
+    "signal",
+    "seed",
+    "data_type",
+    "fold"
+]
+
+# create list to store gene specific results
+gene_auc_list = []
+gene_aupr_list = []
+gene_coef_list = []
+gene_metrics_list = []
+
+# create directory for the gene
+dirname = 'pancancer' if args.use_pancancer else 'single_cancer'
+gene_dir = os.path.join(args.results_dir, dirname, gene_name)
+os.makedirs(gene_dir, exist_ok=True)
+
+# check if gene has been processed already
+# TODO: probably want to add a "resume" option for this in the future
+signal = 'shuffled' if args.shuffle_labels else 'signal'
+check_file = os.path.join(gene_dir,
+                          "{}_{}_{}_coefficients.tsv.gz".format(
+                              gene_name, args.holdout_cancer_type, signal))
+if check_status(check_file):
+    exit()
+
+# process the y matrix for the given gene or pathway
+y_mutation_df = mutation_df.loc[:, gene_name]
+
+# include copy number gains for oncogenes
+# and copy number loss for tumor suppressor genes (TSG)
+include_copy = True
+if classification == "Oncogene":
+    y_copy_number_df = copy_gain_df.loc[:, gene_name]
+elif classification == "TSG":
+    y_copy_number_df = copy_loss_df.loc[:, gene_name]
+else:
+    y_copy_number_df = pd.DataFrame()
+    include_copy = False
+
+y_df = process_y_matrix(
```

Comment:
Might be helpful to add a comment here on what this is returning. Or in the function itself. I had a little trouble following..but I'm also outside of my field on this one